etc/helm/pachyderm/templates/_helpers.tpl

```diff
@@ -24,13 +24,28 @@ LOCAL
 {{- end -}}
 
 {{- define "pachyderm.consoleSecret" -}}
-{{ default (randAlphaNum 32) .Values.console.config.oauthClientSecret }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.console.config.oauthClientSecret "hashSalt" "consoleSecret" "OnlyDefault" .Release.IsUpgrade) }}
 {{- end -}}
 
 {{- define "pachyderm.clusterDeploymentId" -}}
-{{ default (randAlphaNum 32) .Values.pachd.clusterDeploymentID }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.pachd.clusterDeploymentID "hashSalt" "deploymentID" "OnlyDefault" .Release.IsUpgrade) }}
 {{- end -}}
 
 {{- define "pachyderm.enterpriseSecret" -}}
-{{ default (randAlphaNum 32) .Values.pachd.enterpriseSecret }}
+{{ include "defaultOrStableHash" (dict "defaultValue" .Values.pachd.enterpriseSecret "hashSalt" "enterpriseSecret" "OnlyDefault" .Release.IsUpgrade) }}
+{{- end -}}
+
+## if 'defaultValue' isn't defined use the date/time to create a hash
+## truncate the date to the minute and use 'genPrefix' to produce a different hash per useage
+## expects a context containing 'defaultValue' and 'hashSalt' keys
+{{- define "defaultOrStableHash" -}}
+{{- if .defaultValue }}
+    {{- .defaultValue }}
+{{- else }}
+    {{- if .OnlyDefault }}
+        {{- fail "must provide a default value during release upgrade." }}
+    {{- else }}
+        {{- derivePassword 1 "long" (now | toString | trunc 16) .hashSalt "pachyderm" -}}
```

Comment:
Can you add some docs here? (ie, talking about the date truncation, perhaps referencing the article we found this in.